File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/quiz/QuizCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.quiz;

import java.util.List;
import ma.zs.alc.bean.core.quiz.Quiz;
import ma.zs.alc.dao.criteria.core.quiz.QuizCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface QuizCollaboratorService extends  IService<Quiz,QuizCriteria>  {

    List<Quiz> findBySectionId(Long id);
    int deleteBySectionId(Long id);
    long countBySectionCode(String code);




}

File: frontend-bqckup/src/app/shared/service/collaborator/quiz/QuizCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {QuizDto} from '../../../model/quiz/Quiz.model';
import {QuizCriteria} from '../../../criteria/quiz/QuizCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class QuizCollaboratorService extends AbstractService<QuizDto, QuizCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/quiz/';
    }

    public constrcutDto(): QuizDto {
        return new QuizDto();
    }

    public constrcutCriteria(): QuizCriteria {
        return new QuizCriteria();
    }
}
